src/Microsoft.ML.Transforms/Text/WordEmbeddingsTransform.cs

```diff
@@ -324,9 +343,93 @@ public Mapper(WordEmbeddingsTransform parent, Schema inputSchema)
+                //
+                // Symbols:
+                // j: # of features per word in the pretrained model
+                // n: # of input words
+                // X: word input, n tensor
+                // k: # of words in pretrained model (known when transform is created)
+                // S: word labels, k tensor (known when transform is created)
+                // D: word embeddings, k-by-j tensor(known when transform is created)
+                //
+                // X[n]-- > LabelEncoder(classes_strings = S[k], default_int64 = 1)        D[k, j]->Initializer
+                //                              |                                                   |
+                //                             Y[n]                                                 |
+                //                              |                                                   |
+                //                              -------->(indices) Gather (axis= 1)(data) <---------
+                //                                                      |
+                //                                                   W[n, j]
+                //                                                /     |      \
+                //                        ---------------------- /      |       \----------------------
+                //						/                               |                                \
+                //              ReduceMin(axes = [0])         ReduceMean(axes = [0])         ReduceMax(axes = [0])
+                //                      |                                |                                |
+                //                    J[j]                             K[j]                             L[j]
+                //                      |                                |                                |
+                //                       --------------------Concat (axis = 1) ---------------------------
+                //                                                       |
+                //                                                   P[j * 3]
+
+                // Allocate D, a constant tensor
+                var shapeD = new long[] { _parent._currentVocab.GetNumWords(), _parent._currentVocab.Dimension };
+                var tensorD = _parent._currentVocab.WordVectors;
+                var nameD = ctx.AddInitializer(tensorD, shapeD, "WordEmbeddingWeights");
+
+                // Retrieve name of X
+                var nameX = srcVariableName;
+
+                // Do label encoding
+                var nameY = ctx.AddIntermediateVariable(null, "LabelEncodedInput", true);
+                var nodeY = ctx.CreateNode("LabelEncoder", nameX, nameY, ctx.GetNodeName("LabelEncoder"));
+                nodeY.AddAttribute("classes_strings", _parent._currentVocab.GetWordLabels());
+                nodeY.AddAttribute("default_int64", 1);
+
+                // Do gather
```

Comment:
`Do gather` doesn't really provide more information here.
```suggestion
                // Map words to their embedding vectors using Gather
```